### PR TITLE
Version Endpoint

### PR DESCRIPTION
**Pull request by sebbader-sap**
Solves #167

### CODE REVIEW

File: common/common.protocol.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "protocolVersions": [
+       {
+            "version": "1.0",
+            "path": "/some/path/v1" 
+       }
+   ]
+}
+```
+
+This data object must comply to the [JSON Schema](schema/version-schema.json) and the [SHACL Shape](shape/version-shape.ttl). The value of the `version` attribute follows the [Semantic Versioning 2.0.0](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string) and thereby has a numeric order on major, minor, and patch versions.
```
Comment (matgnt) [inline common/common.protocol.md:21]: @sebbader-sap we discussed in the meeting that we would leave out the semantic versioning here. This is to be decided as part of the DSP releasing process. The endpoint itself is not directly affected.

**Review by matgnt — COMMENTED** [review]
comment regarding semantic versioning

File: common/common.protocol.md (hunk 1 above)
Comment (sebbader-sap) [reply common/common.protocol.md]: I have rolled back the commit with the semver part. Please check it again.

**Review by matgnt — APPROVED** [review]
looks good to me.

File: common/common.protocol.md (hunk 1 above)
Comment (matgnt) [reply common/common.protocol.md]: looks good to me

**Review by juliapampus — APPROVED** [review]

File: common/common.protocol.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# General Requirements
+
+## Exposure of Dataspace Protocol Versions
+
+Connectors implementing the Dataspace Protocol might operate on different versions. Therefore, it is necessary that they can discover the supported versions of each other in reliable and unambiguous manner. Each Connector must expose information of at least one Dataspace Protocol Version that it supports. The method expose this information is defined through the protocol bindings and might vary amongst them.
```
Comment (jimmarino) [inline common/common.protocol.md:5]:
```suggestion
Connectors implementing the Dataspace Protocol may operate on different versions. Therefore, it is necessary that they can discover the supported versions of each other reliably and unambiguously. Each Connector must expose information of at least one Dataspace Protocol Version it supports. The specifics of how this information is obtained its defined by specific protocol bindings.
```

File: common/common.protocol.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+In any way, a Connector must respond to a respective request by providing a JSON-LD object containing an array of supported versions with at least one item. The item connects the version tag (`version` attribute) with the absolute URL path segment of the root path for all endpoints of this version. The following example specifies that this Connector offers version `1.0` endpoints at `<host>/some/path/v1`.
```
Comment (jimmarino) [inline common/common.protocol.md:7]:
```suggestion
A Connector must respond to a respective request by providing a JSON-LD object containing an array of supported versions with at least one item. The item connects the version tag (`version` attribute) with the absolute URL path segment of the root path for all endpoints of this version. The following example specifies that this Connector offers version `1.0` endpoints at `<host>/some/path/v1`.
```

File: common/common.protocol.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            "version": "1.0",
+            "path": "/some/path/v1" 
+       }
+   ]
+}
+```
+
+This data object must comply to the [JSON Schema](schema/version-schema.json) and the [SHACL Shape](shape/version-shape.ttl).
+
+The requesting Connector, the one that has received the message with the protocol versions, then must select the according endpoints of the sending Connector as well as configure its own client socket accordingly to the selected version. If the Connector can't identify a matching Dataspace Protocol Version, it must terminate the communication.
```
Comment (jimmarino) [inline common/common.protocol.md:23]:
```suggestion
The requesting Connector may select from the endpoints in the response. If the Connector can't identify a matching Dataspace Protocol Version, it must terminate the communication. 
```

**Review by jimmarino — APPROVED** [review]
I had some wording suggestions, but nothing that changes the meaning of the text.

**Pull request merged**